pkg/api/sharedresource/v1alpha1/share_type.go

```diff
@@ -13,58 +8,120 @@ import (
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// Share is the Schema for the shares API
-type Share struct {
+// SharedResource allows a backing Secret or ConfigMap to be shared across namespaces.
+// Pods can mount the shared Secret or ConfigMap by adding a CSI volume to the pod specification using the
+// "sharedresource.storage.openshift.io" CSI driver and a reference to the SharedResource in the volume attributes:
+//
+// spec:
+//  volumes:
+//  - name: shared-secret
+//    csi:
+//      driver: sharedresource.storage.openshift.io
+//      volumeAttributes:
+//        sharedResource: my-share
+//
+// For the mount to be successful, the pod's service account must be granted permission to 'use' the named SharedResource object
```

Comment:
so starting here @coreydaley was my attempt to take what @deads2k and I discussed in slack in reference to your openshift/api PR and apply "corrections" in the godoc